### PR TITLE
Use PredefinedColorSpace instead of GPUPredefinedColorSpace

### PR DESCRIPTION
**Pull request by Kangz**
An issue indicates that doing this would mean that WebGPU automatically
gets support for new color spaces when additional entries are added to
PredefinedColorSpace. This seems like a good thing for the consistency
of the Web platform given WebGL and 2D canvas already do that. If a
future PredefinedColorSpace value is impossible to support in WebGPU
(which is unlikely), then additional validation can be added to the
spec.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Jun 1, 2022, 8:20 AM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2FKangz%2Fgpuweb%2F93889a304e853350148093e20d4216577abbe006%2Fspec%2Findex.bs&force=1&md-warning=not%20ready)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%232967.)._
</details>

### CODE REVIEW

**Comment by Kangz** [comment]
Thanks! Also do you have any idea what might be causing the action issue? I'm not able to repro locally.

**Comment by kainino0x** [comment]
Jim already figured it out in #2966.

File: spec/index.bs [hunk 1]
```diff
@@ -1700,23 +1700,9 @@ premultiplication) in which the WebGPU numeric values are to be interpreted.
 Each color space is defined over an extended range, if defined by the referenced CSS definitions,
```
Comment (kainino0x) [inline spec/index.bs:1700]:
```suggestion
WebGPU allows all of the color spaces in the {{PredefinedColorSpace}} enum.
Note, each color space is defined over an extended range, as defined by the referenced CSS definitions,
```

**Comment by kainino0x** [comment]
> I'd mostly want to make sure we give any WG members who advocated for the separate color space previously a chance to speak up if they know of concrete implementation problems that would arise from this.

If by "separate color space" you mean defining our own `GPUPredefinedColorSpace`, I'm pretty sure we only have that because I did it to make my changes more minimal (add color space API surface without actually adding any support to the API). So the tacit resolution queue item is "We're adding display-p3 to the list of supported color space now"

**Comment by kainino0x** [comment]
> An issue indicates that doing this would mean that WebGPU automatically
gets support for new color spaces when additional entries are added to
PredefinedColorSpace.

For the record: it was clarified in the meeting that `PredefinedColorSpace` is defined in the HTML spec and is specific to canvases. So it's not an enum of ALL of the color spaces in CSS, just the ones that HTML decided to support for canvases in general, and it makes sense for us to support what other canvas types support.

**Comment by juj** [comment]
Should there be a link added to the spec somewhere to point the reader to where `PredefinedColorSpace` is defined?

**Pull request merged**